virtcontainers/sandbox.go

```diff
@@ -1769,7 +1771,16 @@ func (s *Sandbox) AppendDevice(device api.Device) error {
 	switch device.DeviceType() {
 	case config.VhostUserSCSI, config.VhostUserNet, config.VhostUserBlk, config.VhostUserFS:
 		return s.hypervisor.addDevice(device.GetDeviceInfo().(*config.VhostUserDeviceAttrs), vhostuserDev)
+	case config.DeviceVFIO:
+		vfioDevs := device.GetDeviceInfo().([]*config.VFIODev)
+		for _, d := range vfioDevs {
```

Comment:
Maybe add a check that vfioDevs has only one device.

Reply:
is that check really needed? I don't think so, could you elaborate on this?

Reply:
If the slice wouldn't have 0 element or more than one element, accessing vfioDevs[0] directly seems to be more readable.

Reply:
thanks, I consider that the current code is readable enough, @kata-containers/runtime wdyt?

Reply:
I, personally, am not in favour of special-casing a single device unless there's a specific special-case that would only fit a single device.